gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreUtils.java

```diff
@@ -256,6 +261,29 @@ public static SerDeInfo getSerDeInfo(HiveRegistrationUnit unit) {
     return si;
   }
 
+  /**
+   * Util for detecting if {@param hiveTable} has a non-optional union (aka complex unions) column types. A non optional
+   * union is defined as a uniontype with multiple possible types and none of them are null
+   *
+   * @param hiveTable
+   * @return if hive table contains complex uniontype columns
+   */
+  public static boolean containsNonOptionalUnionTypeColumn(HiveTable hiveTable) {
+    if (!isAvroFormat(hiveTable)) {
+      // All values in ORC are optional / nullable
+      return false;
+    }
+
+    return hiveTable.getColumns().stream()
+        .map(HiveRegistrationUnit.Column::getType)
+        .map(Object::toString)
+        .anyMatch(columnType -> columnType.contains("uniontype") && !columnType.contains("void"));
```

Comment:
I believe this is not a correct definition of the complex union. <null, int, string> is a complex union, but you will not detect that in this case

Reply:
You are correct. Upon verifying the definition in https://docs.google.com/document/d/1Go2NrOoeCKfrDJw8MAZnsMYbE2KZbso1zkSY8cpLzaQ/edit#heading=h.gd2qof50gbzs, it seems like there was some confusion on what is "non-optional" / complex.

The correct definition as described in the document is: 
>Non-optional union: A union type with more than one non-null option type
  e.g. [“int”, “string”] or [“null”, “int”, “string”]
Alternate phrasing:
> A schema where there 's a union type of >= 2 branches which are non null.
>[int, null] - no
[int, string] - yes
[int, string, null] yes